From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Thu, 29 Jun 2017 22:59:46 +0200
Subject: [PATCH] init.d/Makefile.am: create nonexisting directories

install can't assume that all leading directories do already exist.
While this will in most cases be true on normal desktop/server
distributions, cross build systems or systems that build single packages
will build into a fresh hierarchy. Add -D to install, in order to create
the leading directories.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 init.d/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init.d/Makefile.am b/init.d/Makefile.am
index 2f92cb7fdde4..4a6e551f0b51 100644
--- a/init.d/Makefile.am
+++ b/init.d/Makefile.am
@@ -35,9 +35,9 @@ install-exec-hook:
 
 install-data-hook:
 if ENABLE_SYSTEMD_LOOKUP
-	install -p -m644 haveged.service $(DESTDIR)`pkg-config --variable=systemdsystemunitdir systemd`/haveged.service;
+	install -p -D -m644 haveged.service $(DESTDIR)`pkg-config --variable=systemdsystemunitdir systemd`/haveged.service;
 else
-	install -p -m644 haveged.service $(DESTDIR)$(unit_dir)/haveged.service;
+	install -p -D -m644 haveged.service $(DESTDIR)$(unit_dir)/haveged.service;
 endif
 	systemctl enable haveged.service;
 
